### PR TITLE
Implement basic offline downloader

### PR DESCRIPTION
**Pull request by gberaudo**
See https://jira.camptocamp.com/browse/GSLUX-84.

### CODE REVIEW

File: src/offline/Downloader.js [hunk 1]
```diff
@@ -1,12 +1,186 @@
...
+        const tile = {
+          coord: tileCoord,
+          url: tileUrlFunction(tileCoord, ol.has.DEVICE_PIXEL_RATIO, projection)
+        };
+        queueByZ.push(tile);
+      });
+
+      const centerTileCoord = [z, (minX + maxX) / 2, (minY + maxY) / 2];
+      queueByZ.sort((a, b) => magnitude2(a.coord, centerTileCoord) - magnitude2(b.coord, centerTileCoord));
+      queue.push(...queueByZ);
```
Comment (ger-benjamin) [inline src/offline/Downloader.js]: ... ? I don't know this notation, what's this ?

Comment (gberaudo) [reply src/offline/Downloader.js]: This is the rest argument. It means, push every element of the array.

Comment (ger-benjamin) [reply src/offline/Downloader.js]: thx

Comment (fgravin) [reply src/offline/Downloader.js]: called `spread` es6

**Pull request merged**